src/ThemeToggle.svelte

```diff
@@ -0,0 +1,187 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+
+  let theme: "light" | "dark" | undefined = undefined;
```

Comment:
```suggestion
  let theme: Theme | null = null;
  
  enum Theme {
    Light = "light",
    Dark = "dark"
  }
```

I suggest we keep using string enums here, similar to other places in the web ui, and we prefer to use `null` over `undefined`

Reply:
Happy to oblige: ec07fc3fe2e9b0be0f279f65eab5798bafd5fca6.

Just a couple of questions for my own understanding:

1) Is it just a convention or are there actual benefits for using enums? I remember discussing this before in Upstream and I think the outcome was that string literal types had all the same benefits of enums _and_ were simpler and less verbose.

2) What's the thinking behind using `null`? From my point it also makes stuff more verbose. For example, we can't use the optional chaining operator `?.` with component props:

```
// component.svelte
<script>
  export let myFancyProp: string | undefined = undefined;
</script>

// using the component

<Component myFancyProp={someObject?.maybeEmptyKey}>
```

vs

```
// component.svelte
<script>
  export let myFancyProp: string | null = null;
</script>

// using the component

<Component myFancyProp={someObject?.maybeEmptyKey || null}>
```